modelling/src/main/java/org/axonframework/modelling/command/inspection/AnnotatedAggregate.java

```diff
@@ -412,8 +407,17 @@ private Object handle(CommandMessage<?> commandMessage) throws Exception {
         MessageHandlingMember<? super T> handler =
                 inspector.commandHandlers((Class<? extends T>) aggregateRoot.getClass())
                          .filter(mh -> mh.canHandle(commandMessage))
+                         .collect(Collectors.collectingAndThen(Collectors.toList(), Optional::of))
+                         .filter(mhl -> !mhl.isEmpty())
+                         .orElseThrow(() -> new NoHandlerForCommandException(commandMessage))
+                         .stream()
+                         .filter(mh -> mh.unwrap(CommandMessageHandlingMember.class)
+                                         .map(c -> c.canResolve(commandMessage, aggregateRoot))
+                                         .orElse(false))
```

Comment:
I think `orElse(true)` is semantically more correct. We have already filtered on handlers that can handle the command. If we cannot say for sure if the state of the handler is correct to be invoked, we should consider the handler "suitable" and let it throw an exception if it cannot actually handle it.

Reply:
I fixed it following your suggestion, please take a look.